pysumma/simulation.py

```diff
@@ -39,161 +39,165 @@ def __init__(self, executable, filemanager, run_suffix='_pysumma_run'):
                     + self.output_prefix.value + '_'
                     + self.decision_obj.simulStart.value[0:4] + '-'
-                    + self.decision_obj.simulFinsh.value[0:4] + '_' + '1.nc')
+                    + self.decision_obj.simulFinsh.value[0:4] + '_'
+                    + self.run_suffix + '1.nc')
+                #TODO: This could be done via xarray - also eneralized
+                merge_netcdf = ('ncrcat ' + myoutput_list
+                                + ' -O ' + output_name)
+                subprocess.run(merge_netcdf, shell=True)
+                myList = ' '.join(output_list[:])
+                #TODO: This is a bad idea
+                delete_netcdf = 'rm -rf ' + myList
+                subprocess.run(delete_netcdf, shell=True)
 
-            elif specworker_img == 'cuahsi/summa:sopron':
-                # save these paths in the env_vars dictionary
-                # which will be passed to the model
-                env_vars = {'LOCALBASEDIR': self.base_dir,
-                            'MASTERPATH': self.filepath}
-                # define the location we want to mount
-                # these data in the container
-                vol_target = '/tmp/summa'
-                # define the base path of the input data for SUMMA
-                vol_source = self.base_dir
-                # run the container with the arguments specified above
-                res = jobs.run(specworker_img, '-x', vol_source,
-                               vol_target, env_vars)
+            # define output file name as sopron version of summa
+            out_file_path = (
+                self.base_dir + self.output_path.value.split('>')[1]
+                + self.output_prefix.value + '_'
+                + self.decisions.simulStart.value[0:4] + '-'
+                + self.decisions.simulFinsh.value[0:4] + '_' + '1.nc')
 
-                # define output file name as sopron version of summa
-                out_file_path = (
-                    self.base_dir + '/'
-                    + self.output_path.filepath.split('/')[1] + '/'
-                    + self.output_prefix.value + '_output_' + 'timestep.nc')
+        elif specworker_img == 'cuahsi/summa:sopron':
+            out_file_path = (
+                self.base_dir + '/'
+                + self.output_path.filepath.split('/')[1] + '/'
+                + self.output_prefix.value + '_output_' + 'timestep.nc')
+            #TODO: WHy doesn't this run anything?
+        else:
+            #TODO: This error message isn't specific enough
+            raise ValueError('You need to deinfe the '
+                             'exact SUMMA_image_name')
+        return out_file_path
 
-            else:
-                raise ValueError('You need to deinfe the '
-                                 'exact SUMMA_image_name')
+    def exec_docker(self, run_suffix, docker_img):
+
+        self.executable = docker_img
```

Comment:
How about having it so that the function definition is: `def exec_docker(self, run_suffix, docker_img='uwhydro/summa:sopron_2018')`? This way users can still provide their own docker images (There is at least one other research group here at UW that have put together their own SUMMA docker images).

Reply:
Done.